### ISSUE TITLE
[FR] Syncthing-Fork should reliably end syncthing process upon exit

### ISSUE THREAD

**Issue by ramesh45345** [Catfriend1/syncthing-android#290]
Upon exiting Syncthing-Fork, it would be nice if it would optionally clean up or end the syncthing process running in the background (regardless of its state of running properly or hung for whatever reason).

This is especially useful when, sometimes, Syncthing-Fork stays in the "Syncthing is starting..." status (for whatever reason, such as an invalid syncthing configuration, etc), and Syncthing-Fork is exited and re-run, Syncthing-Fork will not run since it knows port 8384 is being used by possibly another syncthing process. If Syncthing-Fork cleans up processes regardless of whether or not they are running successfully, this would eliminate this possibility.

### Steps to Reproduce

1. Open Syncthing-Fork
2. Close Syncthing-Fork
3. Use a task manager to see that syncthing is running in the background.

### Extra Steps to Reproduce

1. Open Syncthing-Fork, and very quickly after opening it, close it (while Syncthing-Fork is still in the "starting" phase).
2. Re-open Syncthing-Fork.
3. Note that Syncthing-Fork cannot start, since the syncthing port (for me, 8384) is busy.

### Version Information
    App Version: 1.0.0.19
    Syncthing Version: v1.0.0
    Android Version: Android 7.1.2 (LineageOS 14.1)
    Device manufacturer: Samsung
    Device model: i747 (actually running lineage_jfltevzw)

### Logs
    Will provide if needed. Should be easy to reproduce (hopefully).

**Comment by Catfriend1** [comment]
Thanks for providing that much detail on the issue. I'll look into in one of my next dev sessions.

**Comment by Catfriend1** [comment]
PR "Don't crash if config got corrupted and inform the user #295" should help to catch corrupted configs and tell this to the user properly.

**Comment by Catfriend1** [comment]
Can you please check and confirm if the new release solves the issue?
https://github.com/Catfriend1/syncthing-android/releases/download/v1.0.0.20/com.github.catfriend1.syncthingandroid_v1.0.0.20_976d4e05.apk

**Comment by Catfriend1** [comment]
It's intended to exit the background syncthing native process when you choose exit from the drawer menu. Could maybe run conditions be a cause for the behavior? I can no longer reproduce the leftover native process since 1.0.0.20 on the emulator. My test was letting Android studio start the app and while that was coming up I stress test like hitted the back or home button (leaves an intact native running at intended) and then did the same with going to the drawer menu quickly and hitting the exit button during yellow starting state. That's why I now suspect the behavior you described may come from run condition toggling?!

**Comment by Catfriend1** [comment]
One of the next releases will have a verbose logging option in the preferences screen to obtain more information in the logs to analyze deeper problems.

**Comment by ramesh45345** [comment]
Alright, I will test with a log from the next release and post that when it becomes available. I can confirm on my phone with the above debug build, and the latest release from the Play Store (v1.0.0.20), that there is still a syncthing-fork process in the background when exiting the application using the "Exit" option in the Syncthing-Fork menu. I am using 3C Toolbox to check for the background process in its task manager, however I think any task manager should show the process. Note that my testing is on my Galaxy S4 device I referenced in the first post.

**Comment by ramesh45345** [comment]
Ok, I did not test this before, and now I am starting to think that the issue really has been solved after my observations.

So I do have remote access enabled. When I exit the app, the website hosted on the phone does disappear. When I re-open the app after leaving the app from a "Syncthing is starting..." state, the notification for the port being busy sometimes does appear, however syncthing is able to start up after waiting for a little while (and the website re-appears on the port). The background process isn't the same as with the previous version, I believe this is just Android keeping the process in memory, as opposed to being active.

Sorry for the trouble, it does seem like the issue is solved, I just hadn't observed what the phone was doing properly. Closing this, looks fine now.
